packages/jellyfish-api-core/__tests__/category/masternode/listMasternodes.test.ts

```diff
@@ -0,0 +1,51 @@
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import { ContainerAdapterClient } from '../../container_adapter_client'
+import { MasternodePagination } from '../../../src/category/masternode'
+
+describe('Masternode', () => {
+  const container = new MasterNodeRegTestContainer()
+  const client = new ContainerAdapterClient(container)
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
+    await container.waitForWalletCoinbaseMaturity()
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  it('should list masternodes', async () => {
+    const masternodes = await client.masternode.listMasternodes()
+
+    for (const masternode in masternodes) {
+      const createdMasternode = masternodes[masternode]
+      expect(typeof createdMasternode.ownerAuthAddress).toStrictEqual('string')
+      expect(typeof createdMasternode.operatorAuthAddress).toStrictEqual('string')
+      expect(typeof createdMasternode.creationHeight).toStrictEqual('number')
+      expect(typeof createdMasternode.resignHeight).toStrictEqual('number')
+      expect(typeof createdMasternode.resignTx).toStrictEqual('string')
+      expect(typeof createdMasternode.banHeight).toStrictEqual('number')
+      expect(typeof createdMasternode.banTx).toStrictEqual('string')
+      expect(typeof createdMasternode.state).toStrictEqual('string')
```

Comment:
The state here perhaps belongs to the MasternodeState enum that you created. While running the test, I get 'ENABLED'